### DISCUSSION TITLE
Cannot access to my code-server IDE via --link

### DISCUSSION THREAD

**Discussion by katsie03** [coder/code-server#3965] [Q&A]
## OS/Web Information

- Web Browser:
- Local OS: Windows 10 Pro
- Remote OS: Ubuntu 20.04
- Remote Architecture: x86_64
- `code-server --version`: 3.11.1

Hello There. 
I couldn't be bothered singing up to slack but I am pretty sure this is a bug.

I am having issues connecting to my Code Server via the `--link` function. After authorising on GitHub it redirects me to [vpsname].cdr.co with the error below. 
![image](https://user-images.githubusercontent.com/50741678/129095681-9395ac3a-e706-4980-a7ba-c116fde64618.png)

Even though `code-server --link` is running within terminal. Which is interesting, I initially though it was a port block or something so I enabled `8080`, `80` and `443`, this did not help.

![image](https://user-images.githubusercontent.com/50741678/129096228-05015f24-5420-4034-a0ae-16bad2e6dc6a.png)

Is this a mistake on my side? I cannot seem to locate the issue.
Thanks

**Comment by jsjoeio** [comment]
This isn't a bug AFAICT. 

You can pass a string after `--link` to change the host name. Example:
```sh
code-server --link guacamole
# Starts code-server on https://guacamole-jsjoeio.cdr.co
```

https://user-images.githubusercontent.com/3806031/129100623-d49b5510-dd50-4e6d-91bc-ab6c1e30ff0d.mov

> I couldn't be bothered singing up to slack but I am pretty sure this is a bug.

No worries at all - you can always open up a Q&A Discussion and if it turns out to be a bug, then you can open a bug report. 

I wonder if it's your VPS? Do you mind updating your description with repro steps? I am happy to try using a similar VPS locally to see if I can reproduce it

**Comment by katsie03** [comment]
Hey. 
Here are my specs to my VPS within the image below for further reference. 
![image](https://user-images.githubusercontent.com/50741678/129109684-25cdaf80-5a9f-462c-ace5-f52de6a1494b.png)

I tried your steps as well passing a string after `--link` which is included within the screen recording down below.
Hope this helps!

https://user-images.githubusercontent.com/50741678/129110649-8b99960a-7c69-4db5-a813-1ea38c8799a0.mp4

**Reply by jsjoeio** [reply]
Hmmm 🤔 I don't suppose you can try disabling the VPS and trying again can you?

Any ideas @code-asher ?

(P.S. changed to Discussion since this seems to be specific to your environment. But if we find a bug, I'll open a new bug report)

**Reply by katsie03** [reply]
> (P.S. changed to Discussion since this seems to be specific to your environment. But if we find a bug, I'll open a new bug report)

Yeah no worries about it.
I have disabled the VPS, rebooted, even reinstalled Ubuntu, i'll try Debian next and see if that does wonders but I am not too sure about that. Will let you know how it goes, I'll also use Azure or GCloud to see if I can reproduce the error within one of their VM Instances.

**Reply by katsie03** [reply]
@jsjoeio Just DMd you on twitter for what happens on a clean installation

**Reply by code-asher** [reply]
I don't have any ideas unfortunately, I think we need some kind of logging somewhere because right now we're just shooting in the dark with these link issues.

**Reply by jsjoeio** [reply]
Oh no! I didn't recognize you and deleted that message 😅😢 Can you repost it here? (Plus then others can chime in to help and we have a public record)
